gpt4all-bindings/python/gpt4all/gpt4all.py

```diff
@@ -203,55 +208,78 @@ def generate(
             n_batch: Number of prompt tokens processed in parallel. Larger values decrease latency but increase resource requirements.
             n_predict: Equivalent to max_tokens, exists for backwards compatibility.
             streaming: If True, this method will instead return a generator that yields tokens as the model generates them.
+            callback: A function with arguments token_id:int and response:str, which receives the tokens from the model as they are generated and stops the generation by returning False.
+            verbose: If True, prints out the final prompt before it is passed to the model.
 
         Returns:
             Either the entire completion or a generator that yields the completion token by token.
         """
         generate_kwargs = dict(
-            prompt=prompt,
             temp=temp,
             top_k=top_k,
             top_p=top_p,
             repeat_penalty=repeat_penalty,
             repeat_last_n=repeat_last_n,
             n_batch=n_batch,
             n_predict=n_predict if n_predict is not None else max_tokens,
+            verbose=verbose,
         )
 
         if self._is_chat_session_activated:
             self.current_chat_session.append({"role": "user", "content": prompt})
-            generate_kwargs['prompt'] = self._format_chat_prompt_template(messages=self.current_chat_session)
+            prompt = self._format_chat_prompt_template(
+                         messages = self.current_chat_session,
+                         default_prompt_header = self.chat_session_header
```

Comment:
Not sure, but isn't the `chat_session_header` supposed to be inserted only at the start of a session? This looks like it goes in there every time a new prompt is started.

Reply:
In the 1.0.2 version, the `_format_chat_prompt_template` used to return the whole chat history each time and so I had to prepend the system prompt every time. In the new fix, 1.0.3, only the last message is sent and so yes, you are right, the system prompt should be included only before the first message.